hyperspy/drawing/markers.py

```
@@ -0,0 +1,875 @@
+            if isinstance(value, da.Array) and value.dtype == object:
+                self.dask_kwargs[key] = self.kwargs[key]
+            elif isinstance(value, da.Array):  # and value.dtype != object:
+                self.kwargs[key] = value.compute()
+            # Patches or verts shouldn't be cast to array
+            elif (
+                isinstance(value, list)
+                and len(value) > 0
+                and not isinstance(value[0], Patch)
+                and not key == "verts"
+            ):
+                self.kwargs[key] = np.array(value)
+            elif isinstance(value, list) and len(value) == 0:
+                self.kwargs[key] = np.array(value)
+
+            if key in ["sizes", "color"] and (
+                not hasattr(value, "__len__") or isinstance(value, str)
+            ):
+                self.kwargs[key] = (value,)
+            if key in ["s"] and isinstance(value, str):
+                self.kwargs[key] = (value,)
+
+        self._cache_dask_chunk_kwargs = {}
+        self._cache_dask_chunk_kwargs_slice = {}
+
+        self.name = self.marker_type
+        # Properties
+        self.collection = None
+        self.collection_class = collection_class
+        self.signal = None
+        self.temp_signal = None
+        self._plot_on_signal = True
+        self.shift = shift
+        self.plot_marker = True
+        self.offsets_transform = offsets_transform
+        self.transform = transform
+
+        # Events
+        self.events = Events()
+        self.events.closed = Event(
+            """
+            Event triggered when a marker is closed.
+
+            Arguments
+            ---------
+            marker : Marker
+                The marker that was closed.
+            """,
+            arguments=["obj"],
+        )
+        self._closing = False
+
+    @property
+    def _is_iterating(self):
+        if self._transform == "relative" or self._offsets_transform == "relative":
```

Do we need to raise an error somewhere when the transform is relative and `offsets`, `sizes`, etc are iterating?

---

Hmm.  I think that it should still work.  There might be cases where you want to have an iterating offsets with a relative plot.  There is no reason that it shouldn't work.